nupic/research/frameworks/vernon/mixins/report_max_accuracy.py

```diff
@@ -0,0 +1,47 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+class ReportMaxAccuracy:
+    """
+    Reports the maximum accuracy and its corresponding epoch during training.
+    """
+    def setup_experiment(self, config):
+        super().setup_experiment(config)
+        self.max_accuracy = 0
+        self.max_accuracy_epoch = 0
+
+    def validate(self):
```

Comment:
`validate` takes in an optional loader, it would be good to support that. You could change this to

```python
def validate(self, *args, **kwargs):
    ret = super().validate(*args, **kwargs)
    ...
```